src/celeritas/track/SimData.hh

```diff
@@ -151,11 +171,37 @@ template<MemSpace M>
 void resize(SimStateData<Ownership::value, M>* data, size_type size)
 {
     CELER_EXPECT(size > 0);
-    StateCollection<SimTrackState, Ownership::value, MemSpace::host> state;
-    std::vector<SimTrackState> initial_state(size);
-    make_builder(&state).insert_back(initial_state.begin(),
-                                     initial_state.end());
-    data->state = state;
+
+    auto resize_state_collection = [size](auto& state_collection) {
+        using value_type = typename std::remove_reference<
+            decltype(state_collection)>::type::value_type;
+        celeritas::StateCollection<value_type, Ownership::value, MemSpace::host>
+            state;
+        std::vector<value_type> initial_state(size);
+        make_builder(&state).insert_back(initial_state.begin(),
+                                         initial_state.end());
+        state_collection = state;
+    };
+    auto resize_state_collection_with_default = [size](auto& state_collection,
+                                                       auto init) {
+        using value_type = typename std::remove_reference<
+            decltype(state_collection)>::type::value_type;
+        celeritas::StateCollection<value_type, Ownership::value, MemSpace::host>
+            state;
+        std::vector<value_type> initial_state(size, init);
+        make_builder(&state).insert_back(initial_state.begin(),
+                                         initial_state.end());
+        state_collection = state;
+    };
```

Comment:
We might be able to use a `resize()` and `fill()` here instead of constructing on the host and copying?

Reply:
Good point, thanks for the feedback!